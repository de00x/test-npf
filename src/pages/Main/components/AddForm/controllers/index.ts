export { default as AddFormService } from './AddForm.service'
export { default as AddFormControllers } from './AddFormControllers'
