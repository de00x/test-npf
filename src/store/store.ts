import { configureStore } from '@reduxjs/toolkit'

// eslint-disable-next-line import/no-named-as-default
import usersReducer from './slices/usersSlice'

export const store = configureStore({
  reducer: {
    users: usersReducer,
  },
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
