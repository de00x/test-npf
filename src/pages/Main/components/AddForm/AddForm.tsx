import { IInputDataProps } from '../../../../types/userInputData.types'
import { MainControllers } from '../../controllers'
import styles from './styles/AddForm.module.scss'
import { FC } from 'react'

export const AddForm: FC<IInputDataProps> = ({ userData, setUserData }): JSX.Element => {
  const { changeName, changeUsername, changeEmail } = MainControllers({ userData, setUserData })

  return (
    <form className={styles.addForm}>
      <input
        type="text"
        maxLength={15}
        placeholder="user"
        value={userData.user}
        onChange={(e) => changeName(e)}
      />
      <input
        type="text"
        maxLength={15}
        placeholder="username"
        value={userData.username}
        onChange={(e) => changeUsername(e)}
      />
      <input
        type="email"
        maxLength={15}
        placeholder="email"
        value={userData.email}
        onChange={(e) => changeEmail(e)}
      />
    </form>
  )
}
