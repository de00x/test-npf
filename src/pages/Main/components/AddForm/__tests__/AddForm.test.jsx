import { render, screen } from '@testing-library/react'
import { Provider } from 'react-redux'

import { store } from '../../../../../store/store'
import { AddForm } from '../AddForm'

jest.mock('axios', () => {
  return {
    __esModule: true,
    default: jest.fn(),
  }
})

// eslint-disable-next-line react/prop-types
const ReduxWrapper = ({ children }) => <Provider store={store}>{children}</Provider>

describe('AddForm component', () => {
  it('AddForm renders', () => {
    render(<AddForm />, { wrapper: ReduxWrapper })
  })

  it('AddForm text renders', () => {
    render(<AddForm />, { wrapper: ReduxWrapper })

    expect(screen.getByText('Add from API')).toBeInTheDocument()
  })

  it('AddForm renders without', () => {
    render(<AddForm />, { wrapper: ReduxWrapper })

    expect(screen.queryByRole('list')).toBeNull()
  })

  it('AddForm snapshot', () => {
    const addForm = render(<AddForm />, { wrapper: ReduxWrapper })

    expect(addForm).toMatchSnapshot()
  })
})
