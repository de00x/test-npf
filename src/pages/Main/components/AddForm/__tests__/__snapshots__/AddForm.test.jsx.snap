// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddForm component AddForm snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="addForm"
      >
        <div
          class="inputContainer"
        >
          <input
            placeholder="name"
            required=""
            type="text"
            value=""
          />
          <input
            placeholder="username"
            required=""
            type="text"
            value=""
          />
          <input
            pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\\\\.[a-z]{2,4}$"
            placeholder="email"
            required=""
            type="email"
            value=""
          />
        </div>
        <div
          class="btnAddFromFormContainer"
        >
          <button
            class="btnDisabled"
            type="submit"
          >
            Add from Form
          </button>
        </div>
      </form>
      <div
        class="btnAddFromApiContainer"
      >
        <button
          class=""
        >
          Add from API
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <form
      class="addForm"
    >
      <div
        class="inputContainer"
      >
        <input
          placeholder="name"
          required=""
          type="text"
          value=""
        />
        <input
          placeholder="username"
          required=""
          type="text"
          value=""
        />
        <input
          pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\\\\.[a-z]{2,4}$"
          placeholder="email"
          required=""
          type="email"
          value=""
        />
      </div>
      <div
        class="btnAddFromFormContainer"
      >
        <button
          class="btnDisabled"
          type="submit"
        >
          Add from Form
        </button>
      </div>
    </form>
    <div
      class="btnAddFromApiContainer"
    >
      <button
        class=""
      >
        Add from API
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
